pkg/statscombiner/statscombiner_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statscombiner
+
+import "testing"
+
+func TestStatCombiner(t *testing.T) {
+	sc := NewStatsCombiner[int](2)
+
+	type testRun struct {
+		Name            string
+		IntervalStats   map[string][]int
+		ExpectedEntries int
+	}
+
+	tests := []testRun{
+		{
+			Name: "one node reports stats",
+			IntervalStats: map[string][]int{
+				"node1": {1},
+			},
+			ExpectedEntries: 1,
+		},
+		{
+			Name:            "no node reports stats, old stats should still be there",
+			IntervalStats:   map[string][]int{},
+			ExpectedEntries: 1,
+		},
+		{
+			Name:            "no node reports stats, old stats should have gotten deleted",
+			IntervalStats:   map[string][]int{},
+			ExpectedEntries: 0,
+		},
+
```

Comment:
```suggestion
```

Reply:
I think having the ExpectedEntries there despite having the default value helps reading the test cases, but I'm fine with removing it.

Reply:
I think it is not needed.